File: 151-reverse-words-in-a-string/reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        Deque<String> q = new ArrayDeque<>();

        int start = -1 ;
        int i = 0 ;

        while (i < s.length())
        {
            if (s.charAt(i) != ' ')
            {
                start = i ;

                while (i < s.length() && s.charAt(i) != ' ')
                {
                    i++ ;
                }

                q.addFirst(s.substring(start , i)) ;
                i--;

            }
            i++ ;
        }
        return String.join(" ",q) ;
        
    }
}

File: 151-reverse-words-in-a-string/reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:

        #return " ".join(s.split()[::-1])

        q = deque()
        
        start = -1 
        i = 0

        while i < len(s):

            if s[i] != " ":

                start = i
                while i <len(s) and s[i]!= " ":
                    i += 1
                q.appendleft(s[start:i])

                i-= 1
            i+= 1
        return " ".join(q)


        